Apps/CocoaPods-FCM/src/View/DashboardView.swift

```diff
@@ -1,8 +1,102 @@
 import CioDataPipelines
+import CioMessagingInApp
 import SwiftUI
+import UIKit
 import UserNotifications
 
+// ObservableObject that tracks the height of a subview
+class HeightTracker: ObservableObject {
+    @Published var subviewHeight: CGFloat = 0
+}
+
+// Used for associating the height tracker with InAppMessageView
+private var heightTrackerKey: UInt8 = 0
+private var maxHeight: CGFloat = 0
+
+extension InAppMessageView {
+    var heightTracker: HeightTracker? {
+        get {
+            objc_getAssociatedObject(self, &heightTrackerKey) as? HeightTracker
+        }
+        set {
+            objc_setAssociatedObject(self, &heightTrackerKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
```

Comment:
what is the need for `objc_getAssociatedObject` and `objc_setAssociatedObject`?

Reply:
Outdated.